src/java.base/share/native/libjli/java.c

```diff
@@ -806,18 +805,22 @@ static void
 SetJvmEnvironment(int argc, char **argv) {
 
     static const char*  NMT_Env_Name    = "NMT_LEVEL_";
+    const char* NMT_Arg_Name = IsJavaArgs() ? "-J-XX:NativeMemoryTracking=" : "-XX:NativeMemoryTracking=";
     int i;
     /* process only the launcher arguments */
     for (i = 0; i < argc; i++) {
         char *arg = argv[i];
         /*
-         * Since this must be a VM flag we stop processing once we see
-         * an argument the launcher would not have processed beyond (such
-         * as -version or -h), or an argument that indicates the following
-         * arguments are for the application (i.e. the main class name, or
-         * the -jar argument).
+         * Java launcher (!IsJavaArgs()):
+         *   Since this must be a VM flag we stop processing once we see
+         *   an argument the launcher would not have processed beyond (such
+         *   as -version or -h), or an argument that indicates the following
+         *   arguments are for the application (i.e. the main class name, or
+         *   the -jar argument).
+         * Non-java launchers:
+         *   All "-J" arguments are translated to VM args (see TranslateApplicationArgs).
```

Comment:
What do you mean by this? The -J args are not "translated" here but later in TranslateApplicationArgs.

For non-Java launchers AFAICS you have to process the entire argv array because you don't know where they may appear in general. So to me the comment should be:

* Other launchers (IsJavaArgs())
*   All arguments have to be scanned to see if it is a -J argument

Reply:
> What do you mean by this? The -J args are not "translated" here but later in TranslateApplicationArgs.

I meant that they are translated in TranslateApplicationArgs. Looks like it's not clear. Updated the comment as you suggested.